pkg/kindsys/kindsysreport/kindsysreport.go

```diff
@@ -0,0 +1,73 @@
+package kindsysreport
+
+import (
+	"cuelang.org/go/cue"
+)
+
+type AttributeWalker struct {
+	seen  map[cue.Value]bool
+	count map[string]int
+}
+
+func (w *AttributeWalker) Count(sch cue.Value, attrs ...string) map[string]int {
+	w.seen = make(map[cue.Value]bool)
+	w.count = make(map[string]int)
+
+	for _, attr := range attrs {
+		w.count[attr] = 0
+	}
+
+	w.walk(cue.MakePath(), sch)
+	return w.count
+}
+
+func (w *AttributeWalker) walk(p cue.Path, v cue.Value) {
+	if w.seen[v] {
+		return
+	}
+
+	w.seen[v] = true
+
+	for attr := range w.count {
+		if found := v.Attribute(attr); found.Err() == nil {
+			w.count[attr]++
+		}
+	}
+
+	switch v.Kind() {
+	case cue.StructKind:
+		// If current cue.Value is a reference to another
+		// definition, we don't want to traverse its fields
+		// individually, because we'll do so for the actual def.
+		if v != cue.Dereference(v) {
+			return
+		}
```

Comment:
@sdboyer Here's what solves the duplicates, as discussed [here](https://github.com/grafana/schematization-and-as-code-project/issues/41#issuecomment-1398802441).

Reply:
Ah yeah, this is an even simpler way to solve it than what i suggested with a `map[cue.Value]bool`.

It also has the benefit that it won't escape the hierarchy of the schema declaration itself - like, if we have an `@grafanamaturity` attribute in the common schemas, it won't get counted in every kind that references it. (Though that may be something we want to do later 🤷🏻 )

LGTM!